lib/board.rb

```diff
@@ -0,0 +1,52 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    reset!
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(position)
+    @cells[position.to_i - 1]
+  end
+
+  def update(position, player)
+    @cells[position.to_i - 1] = player.token
+  end
+
+  def full?
+    #@cells.count {|token| token == "X" || token == "O"} == 9 ? true : false
+    #refactored to:
+    !@cells.any? {|position| position == " "}
+    
+    #can also do:
+    #@cells.all?{|i| i == "X" || i == "O"}
```

Comment:
Yep. I would argue that `all?` is slightly more semantic, but either approach is really fine.